### PR TITLE
Add support for tokens for SMS

### PR DESCRIPTION
**Pull request by kuzmany**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

This PR added support for suggestion tokens similar like you can use in builder

image

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. On production env need `mautic:assets:generate`
3.  On SMS text message field you are able to use `{` + CTRL SPACE to autosuggestion of tokens, included {ownerfield} tokens
4. Then send SMS and see If tokens were replace correctly

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10779"></a>

### CODE REVIEW

**Comment by kuzmany** [comment]
@escopecz I've made basic unit tests? This need more If coverage is still in blocked mode?

File: app/bundles/SmsBundle/Controller/AjaxController.php [hunk 1]
```diff
@@ -60,4 +64,37 @@ protected function getSmsCountStatsAction(Request $request)
...
     }
+
+    /**
+     * Just selected get tokens from email  builder.
+     *
+     * @param string|null $query
+     *
+     * @return array<string,array<int|string>>
+     */
+    protected function getBuilderTokens($query): array
```
Comment (escopecz) [inline app/bundles/SmsBundle/Controller/AjaxController.php:90]: @kuzmany the unit tests look good! Please create also a functional test that will be calling this new endpoint so we can be sure it won't ever be broken.

**Review by npracht — DISMISSED** [review]
This works great:
1. Suggestion of tokens once i type `{`
2. Token replacement on sent

🚀

File: app/bundles/SmsBundle/Controller/AjaxController.php (hunk 1 above)
Comment (mabumusa1) [reply app/bundles/SmsBundle/Controller/AjaxController.php:90]: @kuzmany can you please address this conversation so we can merge this PR? the code is great and can be merged

**Comment by jos0405** [comment]
I was gonna test this, but getting
image

**Comment by escopecz** [comment]
@kuzmany please resolve the conflicts

**Comment by kuzmany** [comment]
@escopecz conflicts resolved

**Comment by kuzmany** [comment]
@escopecz what unit tests we need to add? Included app/bundles/LeadBundle/Tests/EventListener/OwnerSubscriberTest.php

**Comment by escopecz** [comment]
> what unit tests we need to add?

@kuzmany browse https://github.com/mautic/mautic/pull/11864/files for the highlighted lines of code that are not covered by any test.

**Comment by MadlenF** [comment]
> I was gonna test this, but getting image

@kuzmany I wanted to test but still getting the above error that @jos0405 mentioned.

**Comment by RCheesley** [comment]
@MadlenF @jos0405 this seems to be fixed now - ddev env is loading as expected!

**Review by RCheesley — COMMENTED** [review]
When I try to clone an SMS:

`[2023-10-26 16:30:36] mautic.CRITICAL: Uncaught PHP Exception TypeError: "Mautic\SmsBundle\Controller\SmsController::newAction(): Argument #1 ($request) must be of type Symfony\Component\HttpFoundation\Request, Mautic\SmsBundle\Entity\Sms given, called in /var/www/html/app/bundles/SmsBundle/Controller/SmsController.php on line 560" at /var/www/html/app/bundles/SmsBundle/Controller/SmsController.php line 244 {"exception":"[object] (TypeError(code: 0): Mautic\\SmsBundle\\Controller\\SmsController::newAction(): Argument #1 ($request) must be of type Symfony\\Component\\HttpFoundation\\Request, Mautic\\SmsBundle\\Entity\\Sms given, called in /var/www/html/app/bundles/SmsBundle/Controller/SmsController.php on line 560 at /var/www/html/app/bundles/SmsBundle/Controller/SmsController.php:244)"} {"hostname":"mautic-web","pid":2167}
`

Just checking to see if this is a bug with Mautic rather than this PR.

Tested with segment, marketing message and direct send via campaign and the fields were replaced as expected.

**Review by RCheesley — DISMISSED** [review]
The clone bug is Mautic 5 and not this PR (see https://github.com/mautic/mautic/issues/12830) so this PR is good to go 🚀 🎉

**Comment by atowne-mautic** [comment]
A suggestion for future improvement (not worth blocking this much-needed improvement) - on the SMS builder UI, could you please add a bit of text that says something like "To select a token, press ctrl+space." 

We could even use exactly the same as what's available for code mode for emails:
image

**Comment by kuzmany** [comment]
@atowne-mautic good point. I added it below the text

image

**Pull request merged**